### ISSUE TITLE
User authentication

### ISSUE THREAD

**Issue by cayman54** [theupdateframework/tuf#583]
Hi,
Is it possible to authenticate the user that wants to update his software ?
If a user has a valid account he can download updates, but if he doesn't have any account or his account is disabled/expired he cannot access updates.
Thanks

**Comment by vladimir-v-diaz** [comment]
Thank you for opening an issue, and for your interest in TUF.

Is this a feature request to add user authentication to TUF?  Or are you curious how authentication can work alongside TUF?

It is certainly possible to authenticate users and restrict access to a repository.   However, the authentication needs to be enforced by the software update system.  The reason is due to how TUF works, which is to secure the update process using only metadata that is added to the repository.  TUF isn't really an updater, but a specification on how to validate updates.

In other words, a software updater can use TUF to verify whether the information provided by a repository is accurate.

Nevertheless, we do consider and accept requests to make it easier to work with private repositories.  Please have a look at one such proposal: https://github.com/theupdateframework/taps/blob/master/tap5.md#example-restricting-trust-in-a-community-repository-to-a-single-project

**Comment by trishankatdatadog** [comment]
Authentication of users is out of the scope of TUF. However, you might be able to use mutual TLS auth for this purpose.
